src/Bridge/Doctrine/Repository/SeoOverrideRepository.php

```diff
@@ -12,23 +12,31 @@
 namespace Joli\SeoOverride\Bridge\Doctrine\Repository;
 
 use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\NonUniqueResultException;
 use Joli\SeoOverride\Bridge\Doctrine\Entity\SeoOverride;
 
 class SeoOverrideRepository extends EntityRepository
 {
     /**
+     * @throws NonUniqueResultException
+     *
      * @return SeoOverride|null
      */
     public function findOneForPathAndDomain(string $path, string $domainAlias = null)
     {
-        return $this->createQueryBuilder('s')
-             ->andWhere('s.hashedPath = :hashedPath')
-             ->setParameter('hashedPath', sha1($path))
-             ->andWhere('s.domainAlias = :domainAlias')
-             ->setParameter('domainAlias', $domainAlias)
-             ->setMaxResults(1)
-             ->getQuery()
-             ->getOneOrNullResult()
-        ;
+        $qb = $this->createQueryBuilder('s')
+            ->andWhere('s.hashedPath = :hashedPath')
+            ->setParameter('hashedPath', sha1($path));
+
+        if (null === $domainAlias) {
+            $qb->andWhere('s.domainAlias IS NULL');
+        } else {
+            $qb->andWhere('s.domainAlias = :domainAlias')
+                ->setParameter('domainAlias', $domainAlias);
+        }
+
+        return $qb->setMaxResults(1)
+            ->getQuery()
+            ->getOneOrNullResult();
```

Comment:
I think `setMaxResults(1)` is not mandatory since we use `getOneOrNullResult`.

Reply:
Yes it is. If you don't add it, you will get an exception when there are more than one result

Reply:
Yes, it seems logical. Tests are ok 👍

Reply:
@lyrixx Don't see your comment before mine. So it's better to let `setMaxResults(1)` rather than `@throws \Doctrine\ORM\NonUniqueResultException` ?

I mean, is it a better quality code (and if so, why) ?

Reply:
It does not do the same things at all. You can not compare them. What do you want ? A result or an exception ?

Reply:
Put back setMaxResults(1), it adds LIMIT 1 on the SQL query (because we only care about one result).

Reply:
@lyrixx Well, just a robust code. I didn't know it was one or another. But to answer your question, a result.

@damienalexandre I did it, it's like before.

Reply:
I did not see the limit ... Did I miss something?

Reply:
My bad, I thought it was. Now there is the limit.